src/lib/form-field/form-field.ts

```diff
@@ -145,14 +144,16 @@ export class MatFormField extends _MatFormFieldMixinBase
   /** The form-field appearance style. */
   @Input()
   get appearance(): MatFormFieldAppearance {
-    return this._appearance || this._defaultOptions && this._defaultOptions.appearance || 'legacy';
+    return this._appearance;
   }
   set appearance(value: MatFormFieldAppearance) {
-    // If we're switching to `outline` from another appearance, we have to recalculate the gap.
-    if (value !== this._appearance && value === 'outline') {
-      this._initialGapCalculated = false;
+    if (this._appearance !== value && value === 'outline') {
+      // The outline gap cannot update until change detection causes the ngIf updates, causing the
+      // outline elements to show in the dom.  This is a safe action to take because
+      // updateOutlineGap does not cause a change detection cycle to run.  setTimeout is used
+      // because the update needs to be in a a new task rather than the next microtask.
+      setTimeout(() => this.updateOutlineGap());
```

Comment:
does `this._ngZone.onStable.pipe(take(1)).subscribe(...)` work for this? You should also kick it out of the zone